internal/integration/integration_test.go

```diff
@@ -71,6 +71,37 @@ func TestDefaultQueue(t *testing.T) {
 	tc.AssertLogsContain(build, "Hi there")
 }
 
+// This is test case for an interesting edge case!
+// Read more from: https://buildkite-corp.slack.com/archives/C08N9Q51CTC/p1749704326621419?thread_ts=1749704031.661719&cid=C08N9Q51CTC
```

Comment:
Rather than a link to Slack, a couple of words explaining it here would be better both for others (that don't have access to our Slack instance) and us (so we don't have to read through the thread, some of which isn't relevant to  this bug).

Or maybe the comment in `GetScheduledJobs` is enough?